python-threatexchange/threatexchange/exchanges/clients/techagainstterrorism/api.py

```diff
@@ -0,0 +1,152 @@
+        session = requests.Session()
+        session.mount(
+            self.BASE_URL,
+            adapter=TimeoutHTTPAdapter(
+                timeout=60,
+                max_retries=Retry(
+                    total=4,
+                    status_forcelist=[429, 500, 502, 503, 504],
+                    allowed_methods=["HEAD", "GET", "OPTIONS"],
+                    backoff_factor=0.2,  # ~1.5 seconds of retries
+                ),
+            ),
+        )
+
+        if auth_token:
+            session.headers.update({"Authorization": f"Bearer {auth_token}"})
+
+        try:
+            yield session
+        finally:
+            session.close()
+
+    def _get(self, endpoint: str, auth_token: t.Optional[str] = None) -> t.Any:
+        """
+        Perform an HTTP GET request, and return the JSON response payload.
+
+        Same timeouts and retry strategy as `_get_session` above.
+        """
+        with self._get_session(auth_token) as session:
+            url = self.BASE_URL + endpoint
+            response = session.get(url=url)
+            response.raise_for_status()
+            return response.json()
+
+    def _post(self, endpoint: TATEndpoint, data=None) -> t.Any:
+        """
+        Perform an HTTP POST request, and return the JSON response payload.
+
+        Same timeouts and retry strategy as `_get_session` above.
+        """
+        with self._get_session() as session:
+            url = self.BASE_URL + endpoint.value
+            response = session.post(url=url, data=data)
+            response.raise_for_status()
+            return response.json()
+
+    def get_auth_token(self, username: str, password: str) -> t.Optional[str]:
+        """
+        Authenticate with TCAP services and obtain a JWT token
+        """
+
+        if not isinstance(username, str) or not isinstance(password, str):
+            logging.error("Username or password not valid")
+            return None
+
+        logging.info("Authenticating with TCAP: %s", username)
+
+        auth_response = self._post(
+            TATEndpoint.authenticate,
+            data={"username": username, "password": password, "resend": False},
+        )
+        return auth_response.get("token")
+
+    def get_hash_list(
+        self, ideology: str = TATIdeology._all.value
+    ) -> TATHashListResponse:
+        """
+        Get the Hash List JSON file presigned URL ( 5 Minute expiry ) and metadata: TATHashListResponse
+        """
+
+        try:
+            token = self.get_auth_token(self.username, self.password)
+            endpoint = f"{TATEndpoint.hash_list.value}/{ideology}"
+
+            logging.info("Fetching hash list")
+            response = self._get(endpoint, auth_token=token)
+
+            return TATHashListResponse(**response)
+
+        except Exception as exception:
+            logging.error("Failed to get hash list: %s", exception)
+            raise Exception("Failed to fetch Hash List: %s", exception)
```

Comment:
blocking (though I'm fine fixing it on a followup): Instead of wrapping this exception, rethrow the one on stack by just doing 

```
raise
```

Ref: https://docs.python.org/3/tutorial/errors.html#enriching-exceptions-with-notes

Reply:
I was in the code so I made the change. Thanks for the reference!